cmd/service_linux.go

```diff
@@ -0,0 +1,340 @@
+
+// Section defining all the command that we have in secure feature
+var serviceCmd = &cobra.Command{
+	Use:               "service",
+	Short:             "Manage cloudfuse startup process on Linux",
+	Long:              "Manage cloudfuse startup process on Linux",
+	SuggestFor:        []string{"ser", "serv"},
+	Example:           "cloudfuse service install",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return errors.New("missing command options\n\nDid you mean this?\n\tcloudfuse service install\n\nRun 'cloudfuse service --help' for usage")
+	},
+}
+
+var installCmd = &cobra.Command{
+	Use:               "install",
+	Short:             "Installs the startup process for Cloudfuse. Requires elevated permissions.",
+	Long:              "Installs the startup process for Cloudfuse which remounts any active previously active mounts on startup. elevated permissions.",
+	SuggestFor:        []string{"ins", "inst"},
+	Example:           "cloudfuse service install",
+	FlagErrorHandling: cobra.ExitOnError,
+	Args:              cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// 1. get the cloudfuse.service file from the setup folder and collect relevant data (user, mount, config)
+
+		// get current dir
+		dir, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("error: [%s]", err.Error())
+		}
+
+		// assumes dir is in cloudfuse repo dir
+		serviceData, err := collectServiceData(fmt.Sprintf("%s/setup/cloudfuse.service", dir))
+		if err != nil {
+			return fmt.Errorf("error collecting data from cloudfuse.service file due to the following error: [%s]", err)
+		}
+
+		// TODO: get arguments of cloudfuse service install and pass that to be values written to the service file's config file and mount point paths.
+		mountPath := args[0]
+		configPath := args[1]
+
+		mountExists := common.DirectoryExists(mountPath)
+		if !mountExists {
+			return fmt.Errorf("error, the mount path provided does not exist")
+			// TODO: add useage output upon failure with input
+		}
+		isDirEmpty := common.IsDirectoryEmpty(mountPath)
+		if !isDirEmpty {
+			return fmt.Errorf("error, the mount path provided is not empty")
+			// TODO: add useage output upon failure with input
+		}
+
+		configExists := common.DirectoryExists(configPath)
+		if !configExists {
+			return fmt.Errorf("error, the configfile path provided does not exist") // TODO: add useage output upon failure with input
+		}
+
+		err = modifySericeFile(mountPath, dir)
+		if err != nil {
+			return fmt.Errorf("error when attempting to write defaults into service file: [%s]", err.Error())
+		}
+
+		err = modifySericeFile(configPath, dir)
+		if err != nil {
+			return fmt.Errorf("error when attempting to write defaults into service file: [%s]", err.Error())
+		}
+
+		// 2. retrieve the newUser account from cloudfuse.service file and create it if it doesn't exist
+
+		serviceUser := serviceData["User"]
+		setUser(serviceUser)
```

Comment:
On Linux I don't think we need a dedicated user.

Reply:
While failing to specify a user with the appropriate permissions doesn't stop the cloudfuse service from running and working, it opens up a large security risk involving the default permissions that are used when the service is run.

Reply:
Sorry, you're right. I forgot that on Linux, adding a user is exactly how this is done.